vuu-ui/packages/vuu-shell/src/layout-management/useLayoutManager.tsx

```diff
@@ -1,46 +1,65 @@
 import React, { useState, useCallback, useContext, useEffect } from "react";
-import { getLocalEntity } from "@finos/vuu-filters";
 import { LayoutJSON, LocalLayoutPersistenceManager } from "@finos/vuu-layout";
 import { LayoutMetadata } from "./layoutTypes";
+import { defaultLayout } from "@finos/vuu-layout/";
 
 const persistenceManager = new LocalLayoutPersistenceManager();
 
 export const LayoutManagementContext = React.createContext<{
   layoutMetadata: LayoutMetadata[],
-  saveLayout: (n: Omit<LayoutMetadata, "id">) => void
-}>({ layoutMetadata: [], saveLayout: () => { } })
+  saveLayout: (n: Omit<LayoutMetadata, "id">) => void,
+  currentLayout: LayoutJSON,
+  saveCurrentLayout: (layout: LayoutJSON) => void,
+  loadLayoutById: (id: string) => void
+}>({
+  layoutMetadata: [],
+  saveLayout: () => { },
+  currentLayout: defaultLayout,
+  saveCurrentLayout: () => { },
+  loadLayoutById: () => defaultLayout
+})
 
-export const LayoutManagementProvider = (props: {
-    children: JSX.Element | JSX.Element[]
-  }) => {
+type LayoutManagementProviderProps = {
+  children: JSX.Element | JSX.Element[]
+}
+
+export const LayoutManagementProvider = (props: LayoutManagementProviderProps) => {
   const [layoutMetadata, setLayoutMetadata] = useState<LayoutMetadata[]>([]);
+  const [currentLayout, setCurrentLayout] = useState<LayoutJSON>(defaultLayout);
 
   useEffect(() => {
-    const loadedMetadata = persistenceManager.loadMetadata();
-    setLayoutMetadata(loadedMetadata || [])
+    Promise.all([persistenceManager.loadMetadata(), persistenceManager.loadTempLayout()])
+      .then(([metadata, layout]) => {
+        setLayoutMetadata(metadata)
+        setCurrentLayout(layout);
+      })
```

Comment:
Do we need to use `Promise.all` here, or would it be better to load temp layout and metadata independently? If there's a problem loading the metadata, we wouldn't want that to stop us from setting the current layout.

Reply:
thats a good point. these shouldnt be dependent on each other. Thanks Peter :)